ggml/src/ggml-opt.cpp

```diff
@@ -316,8 +348,14 @@ static void ggml_opt_build(ggml_opt_context_t opt_ctx) {
     GGML_ASSERT(opt_ctx->ctx_compute && "no compute context set, either use static graphs or set one with ggml_opt_prepare_alloc");
     GGML_ASSERT((!opt_ctx->static_graphs || opt_ctx->inputs->data) && "when using static graphs the inputs must be allocated statically");
 
+    const enum ggml_opt_optimizer_type optimizer = opt_ctx->optimizer;
+    const bool has_moment = !opt_ctx->grad_m.empty();
+    const bool need_moment = optimizer == GGML_OPT_OPTIMIZER_TYPE_ADAMW;
+    const bool moment =
+        opt_ctx->build_type_alloc == GGML_OPT_BUILD_TYPE_OPT && (need_moment || has_moment);
+
```

Comment:
```suggestion
    const bool moment = opt_ctx->build_type_alloc == GGML_OPT_BUILD_TYPE_OPT &&  opt_ctx->optimizer == GGML_OPT_OPTIMIZER_TYPE_ADAMW;
```

Sorry, I didn't look at the code again before writing my last comment. This is what I meant the logic to be.